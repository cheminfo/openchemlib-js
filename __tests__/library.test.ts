import { assert, describe, expect, test } from 'vitest';

import debugOCL from '../lib/index.debug.js';
import OCL from '../lib/index.js';

const allAPI = Object.keys(OCL).sort();

test('debug build should have the same exports', () => {
  expect(allAPI).toStrictEqual(Object.keys(debugOCL).sort());
});

test('top-level API', () => {
  expect(allAPI).toMatchSnapshot();
  for (const api of allAPI) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    assert((OCL as any)[api], `Missing top-level API: ${api}`);
  }
});

describe('class prototypes', () => {
  for (const key of allAPI) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const api = (OCL as any)[key];
    if (typeof api === 'function') {
      test(`static properties of ${key}`, () => {
        expect(getFilteredKeys(api)).toMatchSnapshot();
      });
      if (typeof api.prototype === 'object') {
        test(`prototype properties of ${key}`, () => {
          expect(getFilteredKeys(api.prototype)).toMatchSnapshot();
        });
      }
    }
  }
});

function getFilteredKeys(obj: Record<string, unknown>) {
  return Object.keys(obj)
    .filter((key) => {
      // Filter out GWT-specific properties.
      return key.length > 2;
    })
    .sort();
}
