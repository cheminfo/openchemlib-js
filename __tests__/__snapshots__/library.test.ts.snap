// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`class prototypes > prototype properties of Canonizer 1`] = `
[
  "getCanMolecule",
  "getEncodedCoordinates",
  "getEncodedMapping",
  "getFinalRank",
  "getGraphAtoms",
  "getGraphIndexes",
  "getIDCode",
  "getSymmetryRank",
  "getSymmetryRanks",
  "hasCIPParityDistinctionProblem",
  "invalidateCoordinates",
  "normalizeEnantiomer",
  "setParities",
]
`;

exports[`class prototypes > prototype properties of CanonizerUtil 1`] = `[]`;

exports[`class prototypes > prototype properties of CanvasEditor 1`] = `[]`;

exports[`class prototypes > prototype properties of ConformerGenerator 1`] = `
[
  "getConformerCount",
  "getNextConformerAsMolecule",
  "getOneConformerAsMolecule",
  "getPotentialConformerCount",
  "initializeConformers",
  "molecules",
]
`;

exports[`class prototypes > prototype properties of DrugScoreCalculator 1`] = `[]`;

exports[`class prototypes > prototype properties of DruglikenessPredictor 1`] = `
[
  "assessDruglikeness",
  "getDetail",
  "getDruglikenessString",
]
`;

exports[`class prototypes > prototype properties of ForceFieldMMFF94 1`] = `
[
  "getTotalEnergy",
  "minimise",
  "size",
]
`;

exports[`class prototypes > prototype properties of Molecule 1`] = `
[
  "addAtom",
  "addBond",
  "addFragment",
  "addImplicitHydrogens",
  "addMissingChirality",
  "addMolecule",
  "addOrChangeAtom",
  "addOrChangeBond",
  "addRing",
  "addRingToAtom",
  "addRingToBond",
  "addSubstituent",
  "calculateTorsion",
  "canonizeCharge",
  "changeAtom",
  "changeAtomCharge",
  "changeBond",
  "clear",
  "convertStereoBondsToSingleBonds",
  "copyAtom",
  "copyBond",
  "copyMolecule",
  "copyMoleculeByAtoms",
  "copyMoleculeByBonds",
  "copyMoleculeProperties",
  "deleteAtom",
  "deleteAtomOrBond",
  "deleteAtoms",
  "deleteBond",
  "deleteBondAndSurrounding",
  "deleteMarkedAtomsAndBonds",
  "deleteMolecule",
  "deleteSelectedAtoms",
  "ensureHelperArrays",
  "findAlleneCenterAtom",
  "findAlleneEndAtom",
  "findAtom",
  "findBINAPChiralityBond",
  "findBond",
  "findRingSystem",
  "getAbsoluteAtomParity",
  "getAbsoluteBondParity",
  "getAllAtoms",
  "getAllBonds",
  "getAllConnAtoms",
  "getAllConnAtomsPlusMetalBonds",
  "getAllHydrogens",
  "getAromaticRingCount",
  "getAtomAbnormalValence",
  "getAtomCIPParity",
  "getAtomCharge",
  "getAtomColor",
  "getAtomCustomLabel",
  "getAtomESRGroup",
  "getAtomESRType",
  "getAtomLabel",
  "getAtomList",
  "getAtomListString",
  "getAtomMapNo",
  "getAtomMass",
  "getAtomParity",
  "getAtomPi",
  "getAtomPreferredStereoBond",
  "getAtomQueryFeaturesObject",
  "getAtomRadical",
  "getAtomRingBondCount",
  "getAtomRingCount",
  "getAtomRingSize",
  "getAtomX",
  "getAtomY",
  "getAtomZ",
  "getAtomicNo",
  "getAtoms",
  "getAverageBondLength",
  "getAverageTopologicalAtomDistance",
  "getBond",
  "getBondAngle",
  "getBondAtom",
  "getBondBridgeMaxSize",
  "getBondBridgeMinSize",
  "getBondCIPParity",
  "getBondESRGroup",
  "getBondESRType",
  "getBondLength",
  "getBondOrder",
  "getBondParity",
  "getBondPreferredStereoBond",
  "getBondQueryFeatures",
  "getBondQueryFeaturesObject",
  "getBondRingSize",
  "getBondType",
  "getBondTypeSimple",
  "getBonds",
  "getBounds",
  "getCanonizedIDCode",
  "getChiralText",
  "getChirality",
  "getCompactCopy",
  "getConnAtom",
  "getConnAtoms",
  "getConnBond",
  "getConnBondOrder",
  "getDefaultMaxValenceUncharged",
  "getDiastereotopicAtomIDs",
  "getElectronValenceCorrection",
  "getExcludedNeighbourCount",
  "getExplicitHydrogens",
  "getFinalRanks",
  "getFisherProjectionParity",
  "getFragmentAtoms",
  "getFragmentNumbers",
  "getFragments",
  "getFreeValence",
  "getHandleHydrogenMap",
  "getHelperArrayStatus",
  "getHoseCodes",
  "getIDCode",
  "getIDCodeAndCoordinates",
  "getIDCoordinates",
  "getImplicitHigherValence",
  "getImplicitHydrogens",
  "getIndex",
  "getLowestFreeValence",
  "getMaxAtoms",
  "getMaxBonds",
  "getMaxValence",
  "getMaxValenceUncharged",
  "getMetalBondedConnAtoms",
  "getMolecularFormula",
  "getMoleculeColor",
  "getMolweight",
  "getName",
  "getNonHydrogenNeighbourCount",
  "getNumberOfHydrogens",
  "getOCL",
  "getOccupiedValence",
  "getPath",
  "getPathBonds",
  "getPathLength",
  "getRingSet",
  "getRotatableBondCount",
  "getStereoBond",
  "getStereoCenterCount",
  "getStereoProblem",
  "getSubstituent",
  "getSubstituentSize",
  "getSymmetryRank",
  "getZNeighbour",
  "invalidateHelperArrays",
  "inventCoordinates",
  "is3D",
  "isAlkaliMetal",
  "isAllylicAtom",
  "isAmideTypeBond",
  "isAromaticAtom",
  "isAromaticBond",
  "isAtomConfigurationUnknown",
  "isAtomMarkedForDeletion",
  "isAtomParityPseudo",
  "isAtomStereoCenter",
  "isAutoMappedAtom",
  "isBINAPChiralityBond",
  "isBondBackgroundHilited",
  "isBondBridge",
  "isBondForegroundHilited",
  "isBondMarkedForDeletion",
  "isBondParityPseudo",
  "isBondParityUnknownOrNone",
  "isChalcogene",
  "isDelocalizedBond",
  "isEarthAlkaliMetal",
  "isElectronegative",
  "isElectropositive",
  "isFlatNitrogen",
  "isFragment",
  "isHalogene",
  "isMarkedAtom",
  "isMetalAtom",
  "isNaturalAbundance",
  "isNitrogenFamily",
  "isOrganicAtom",
  "isPseudoRotatableBond",
  "isPurelyOrganic",
  "isRingAtom",
  "isRingBond",
  "isSelectedAtom",
  "isSelectedBond",
  "isSimpleHydrogen",
  "isSmallRingAtom",
  "isSmallRingBond",
  "isStabilizedAtom",
  "isStereoBond",
  "markAtomForDeletion",
  "markBondForDeletion",
  "normalizeAmbiguousBonds",
  "removeAtomColors",
  "removeAtomCustomLabels",
  "removeAtomMarkers",
  "removeAtomSelection",
  "removeBondHiliting",
  "removeExplicitHydrogens",
  "removeQueryFeatures",
  "renumberESRGroups",
  "scaleCoords",
  "setAllAtoms",
  "setAllBonds",
  "setAssignParitiesToNitrogen",
  "setAtomAbnormalValence",
  "setAtomCIPParity",
  "setAtomCharge",
  "setAtomColor",
  "setAtomConfigurationUnknown",
  "setAtomCustomLabel",
  "setAtomESR",
  "setAtomList",
  "setAtomMapNo",
  "setAtomMarker",
  "setAtomMass",
  "setAtomParity",
  "setAtomQueryFeature",
  "setAtomRadical",
  "setAtomSelection",
  "setAtomX",
  "setAtomY",
  "setAtomZ",
  "setAtomicNo",
  "setBondAtom",
  "setBondBackgroundHiliting",
  "setBondCIPParity",
  "setBondESR",
  "setBondForegroundHiliting",
  "setBondOrder",
  "setBondParity",
  "setBondParityUnknownOrNone",
  "setBondQueryFeature",
  "setBondType",
  "setChirality",
  "setFragment",
  "setHelperValidity",
  "setHydrogenProtection",
  "setMaxAtoms",
  "setMaxBonds",
  "setMoleculeColor",
  "setName",
  "setParitiesValid",
  "setStereoBondFromAtomParity",
  "setStereoBondFromBondParity",
  "setStereoBondsFromParity",
  "setToRacemate",
  "setUnknownParitiesToExplicitlyUnknown",
  "shareSameFragment",
  "stripIsotopInfo",
  "stripSmallFragments",
  "stripStereoInformation",
  "suggestBondType",
  "supportsImplicitHydrogen",
  "swapAtoms",
  "swapBonds",
  "toIsomericSmiles",
  "toMolfile",
  "toMolfileV3",
  "toSVG",
  "toSmarts",
  "toSmiles",
  "translateCoords",
  "validate",
  "zoomAndRotate",
  "zoomAndRotateInit",
]
`;

exports[`class prototypes > prototype properties of MoleculeProperties 1`] = `[]`;

exports[`class prototypes > prototype properties of ReactionEncoder 1`] = `[]`;

exports[`class prototypes > prototype properties of Reactor 1`] = `
[
  "getProducts",
  "setReactant",
]
`;

exports[`class prototypes > prototype properties of Resources 1`] = `[]`;

exports[`class prototypes > prototype properties of SDFileParser 1`] = `
[
  "getField",
  "getFieldData",
  "getFieldNames",
  "getMolecule",
  "getNextFieldData",
  "getNextMolFile",
  "next",
]
`;

exports[`class prototypes > prototype properties of SSSearcher 1`] = `
[
  "isFragmentInMolecule",
  "setFragment",
  "setMol",
  "setMolecule",
]
`;

exports[`class prototypes > prototype properties of SSSearcherWithIndex 1`] = `
[
  "createIndex",
  "isFragmentInMolecule",
  "setFragment",
  "setMolecule",
]
`;

exports[`class prototypes > prototype properties of SmilesParser 1`] = `
[
  "getSmartsWarning",
  "parseMolecule",
  "parseReaction",
  "setRandomSeed",
]
`;

exports[`class prototypes > prototype properties of ToxicityPredictor 1`] = `
[
  "assessRisk",
  "getDetail",
]
`;

exports[`class prototypes > prototype properties of Transformer 1`] = `
[
  "applyTransformation",
  "setMolecule",
]
`;

exports[`class prototypes > prototype properties of Util 1`] = `[]`;

exports[`class prototypes > prototype properties of registerCustomElement 1`] = `[]`;

exports[`class prototypes > static properties of Canonizer 1`] = `[]`;

exports[`class prototypes > static properties of CanonizerUtil 1`] = `
[
  "BACKBONE",
  "NORMAL",
  "NOSTEREO",
  "NOSTEREO_TAUTOMER",
  "TAUTOMER",
  "getIDCode",
]
`;

exports[`class prototypes > static properties of CanvasEditor 1`] = `[]`;

exports[`class prototypes > static properties of ConformerGenerator 1`] = `
[
  "STRATEGY_ADAPTIVE_RANDOM",
  "STRATEGY_LIKELY_RANDOM",
  "STRATEGY_LIKELY_SYSTEMATIC",
  "STRATEGY_PURE_RANDOM",
]
`;

exports[`class prototypes > static properties of DrugScoreCalculator 1`] = `
[
  "calculate",
]
`;

exports[`class prototypes > static properties of DruglikenessPredictor 1`] = `
[
  "DRUGLIKENESS_UNKNOWN",
]
`;

exports[`class prototypes > static properties of ForceFieldMMFF94 1`] = `
[
  "MMFF94",
  "MMFF94S",
  "MMFF94SPLUS",
]
`;

exports[`class prototypes > static properties of Molecule 1`] = `
[
  "CANONIZER_ASSIGN_PARITIES_TO_TETRAHEDRAL_N",
  "CANONIZER_CONSIDER_DIASTEREOTOPICITY",
  "CANONIZER_CONSIDER_ENANTIOTOPICITY",
  "CANONIZER_CONSIDER_STEREOHETEROTOPICITY",
  "CANONIZER_COORDS_ARE_3D",
  "CANONIZER_CREATE_PSEUDO_STEREO_GROUPS",
  "CANONIZER_CREATE_SYMMETRY_RANK",
  "CANONIZER_DISTINGUISH_RACEMIC_OR_GROUPS",
  "CANONIZER_ENCODE_ATOM_CUSTOM_LABELS",
  "CANONIZER_ENCODE_ATOM_CUSTOM_LABELS_WITHOUT_RANKING",
  "CANONIZER_ENCODE_ATOM_SELECTION",
  "CANONIZER_NEGLECT_ANY_STEREO_INFORMATION",
  "CANONIZER_TIE_BREAK_FREE_VALENCE_ATOMS",
  "FISCHER_PROJECTION_LIMIT",
  "FISCHER_PROJECTION_RING_LIMIT",
  "STEREO_ANGLE_LIMIT",
  "VALIDATION_ERRORS_STEREO",
  "VALIDATION_ERROR_AMBIGUOUS_CONFIGURATION",
  "VALIDATION_ERROR_ESR_CENTER_UNKNOWN",
  "VALIDATION_ERROR_OVER_UNDER_SPECIFIED",
  "cAtomCIPParityNone",
  "cAtomCIPParityProblem",
  "cAtomCIPParityRorM",
  "cAtomCIPParitySorP",
  "cAtomColorBlue",
  "cAtomColorDarkGreen",
  "cAtomColorDarkRed",
  "cAtomColorGreen",
  "cAtomColorMagenta",
  "cAtomColorNone",
  "cAtomColorOrange",
  "cAtomColorRed",
  "cAtomLabel",
  "cAtomParity1",
  "cAtomParity2",
  "cAtomParityIsPseudo",
  "cAtomParityNone",
  "cAtomParityUnknown",
  "cAtomQFAny",
  "cAtomQFAromState",
  "cAtomQFAromStateBits",
  "cAtomQFAromStateShift",
  "cAtomQFAromatic",
  "cAtomQFCharge",
  "cAtomQFChargeBits",
  "cAtomQFChargeShift",
  "cAtomQFENeighbourBits",
  "cAtomQFENeighbourShift",
  "cAtomQFENeighbours",
  "cAtomQFExcludeGroup",
  "cAtomQFFlatNitrogen",
  "cAtomQFHeteroAromatic",
  "cAtomQFHydrogen",
  "cAtomQFHydrogenBits",
  "cAtomQFHydrogenShift",
  "cAtomQFIsNotStereo",
  "cAtomQFIsStereo",
  "cAtomQFMatchStereo",
  "cAtomQFMoreNeighbours",
  "cAtomQFNarrowing",
  "cAtomQFNeighbourBits",
  "cAtomQFNeighbourShift",
  "cAtomQFNeighbours",
  "cAtomQFNewRingSize",
  "cAtomQFNewRingSizeBits",
  "cAtomQFNewRingSizeShift",
  "cAtomQFNoMoreNeighbours",
  "cAtomQFNoOfBits",
  "cAtomQFNot0ENeighbours",
  "cAtomQFNot0Hydrogen",
  "cAtomQFNot0Neighbours",
  "cAtomQFNot0PiElectrons",
  "cAtomQFNot1ENeighbour",
  "cAtomQFNot1Hydrogen",
  "cAtomQFNot1Neighbour",
  "cAtomQFNot1PiElectron",
  "cAtomQFNot2ENeighbours",
  "cAtomQFNot2Hydrogen",
  "cAtomQFNot2Neighbours",
  "cAtomQFNot2PiElectrons",
  "cAtomQFNot2RingBonds",
  "cAtomQFNot3ENeighbours",
  "cAtomQFNot3Hydrogen",
  "cAtomQFNot3Neighbours",
  "cAtomQFNot3RingBonds",
  "cAtomQFNot4ENeighbours",
  "cAtomQFNot4Neighbours",
  "cAtomQFNot4RingBonds",
  "cAtomQFNotAromatic",
  "cAtomQFNotChain",
  "cAtomQFNotCharge0",
  "cAtomQFNotChargeNeg",
  "cAtomQFNotChargePos",
  "cAtomQFPiElectronBits",
  "cAtomQFPiElectronShift",
  "cAtomQFPiElectrons",
  "cAtomQFRingSize0",
  "cAtomQFRingSize3",
  "cAtomQFRingSize4",
  "cAtomQFRingSize5",
  "cAtomQFRingSize6",
  "cAtomQFRingSize7",
  "cAtomQFRingSizeLarge",
  "cAtomQFRingState",
  "cAtomQFRingStateBits",
  "cAtomQFRingStateShift",
  "cAtomQFRxnParityBits",
  "cAtomQFRxnParityHint",
  "cAtomQFRxnParityInvert",
  "cAtomQFRxnParityRacemize",
  "cAtomQFRxnParityRetain",
  "cAtomQFRxnParityShift",
  "cAtomQFSimpleFeatures",
  "cAtomQFSmallRingSize",
  "cAtomQFSmallRingSizeBits",
  "cAtomQFSmallRingSizeShift",
  "cAtomQFStereoState",
  "cAtomQFStereoStateBits",
  "cAtomQFStereoStateShift",
  "cAtomRadicalState",
  "cAtomRadicalStateD",
  "cAtomRadicalStateNone",
  "cAtomRadicalStateS",
  "cAtomRadicalStateShift",
  "cAtomRadicalStateT",
  "cAtomValence",
  "cBondCIPParityEorP",
  "cBondCIPParityNone",
  "cBondCIPParityProblem",
  "cBondCIPParityZorM",
  "cBondParityEor1",
  "cBondParityNone",
  "cBondParityUnknown",
  "cBondParityZor2",
  "cBondQFAllFeatures",
  "cBondQFAromState",
  "cBondQFAromStateBits",
  "cBondQFAromStateShift",
  "cBondQFAromatic",
  "cBondQFBondTypes",
  "cBondQFBondTypesBits",
  "cBondQFBondTypesShift",
  "cBondQFBridge",
  "cBondQFBridgeBits",
  "cBondQFBridgeMin",
  "cBondQFBridgeMinBits",
  "cBondQFBridgeMinShift",
  "cBondQFBridgeShift",
  "cBondQFBridgeSpan",
  "cBondQFBridgeSpanBits",
  "cBondQFBridgeSpanShift",
  "cBondQFDelocalized",
  "cBondQFDouble",
  "cBondQFMatchFormalOrder",
  "cBondQFMatchStereo",
  "cBondQFMetalLigand",
  "cBondQFNarrowing",
  "cBondQFNoOfBits",
  "cBondQFNotAromatic",
  "cBondQFNotRing",
  "cBondQFQuadruple",
  "cBondQFQuintuple",
  "cBondQFRareBondTypes",
  "cBondQFRareBondTypesBits",
  "cBondQFRareBondTypesShift",
  "cBondQFRing",
  "cBondQFRingSize",
  "cBondQFRingSizeBits",
  "cBondQFRingSizeShift",
  "cBondQFRingState",
  "cBondQFRingStateBits",
  "cBondQFRingStateShift",
  "cBondQFSimpleFeatures",
  "cBondQFSingle",
  "cBondQFTriple",
  "cBondTypeCross",
  "cBondTypeDeleted",
  "cBondTypeDelocalized",
  "cBondTypeDouble",
  "cBondTypeDown",
  "cBondTypeIncreaseOrder",
  "cBondTypeMaskSimple",
  "cBondTypeMaskStereo",
  "cBondTypeMetalLigand",
  "cBondTypeQuadruple",
  "cBondTypeQuintuple",
  "cBondTypeSingle",
  "cBondTypeTriple",
  "cBondTypeUp",
  "cChiralityDiastereomers",
  "cChiralityEpimers",
  "cChiralityIsomerCountMask",
  "cChiralityKnownEnantiomer",
  "cChiralityMeso",
  "cChiralityNotChiral",
  "cChiralityRacemic",
  "cChiralityUnknown",
  "cChiralityUnknownEnantiomer",
  "cCommonOxidationState",
  "cDefaultAVBL",
  "cDefaultAllowedPseudoAtoms",
  "cDefaultAtomValence",
  "cESRGroupBits",
  "cESRMaxGroups",
  "cESRTypeAbs",
  "cESRTypeAnd",
  "cESRTypeOr",
  "cHelperAll",
  "cHelperBitCIP",
  "cHelperBitIncludeNitrogenParities",
  "cHelperBitNeighbours",
  "cHelperBitParities",
  "cHelperBitRings",
  "cHelperBitRingsSimple",
  "cHelperBitSymmetrySimple",
  "cHelperBitSymmetryStereoHeterotopicity",
  "cHelperBitsStereo",
  "cHelperCIP",
  "cHelperNeighbours",
  "cHelperNone",
  "cHelperParities",
  "cHelperRings",
  "cHelperRingsSimple",
  "cHelperSymmetrySimple",
  "cHelperSymmetryStereoHeterotopicity",
  "cMaxAtomicNo",
  "cMaxConnAtoms",
  "cMoleculeColorDefault",
  "cMoleculeColorNeutral",
  "cPseudoAtomA",
  "cPseudoAtomAttachmentPoint",
  "cPseudoAtomPolymer",
  "cPseudoAtomR",
  "cPseudoAtomX",
  "cPseudoAtomsAGroups",
  "cPseudoAtomsAll",
  "cPseudoAtomsAminoAcids",
  "cPseudoAtomsHydrogenIsotops",
  "cPseudoAtomsRGroups",
  "cRoundedMass",
  "fromIDCode",
  "fromMolfile",
  "fromMolfileWithAtomMap",
  "fromSmiles",
  "getAngle",
  "getAngleDif",
  "getAtomicNoFromLabel",
  "getDefaultAverageBondLength",
  "isAtomicNoElectronegative",
  "isAtomicNoElectropositive",
  "setDefaultAverageBondLength",
]
`;

exports[`class prototypes > static properties of MoleculeProperties 1`] = `[]`;

exports[`class prototypes > static properties of ReactionEncoder 1`] = `
[
  "CATALYST_DELIMITER",
  "INCLUDE_ALL",
  "INCLUDE_CATALYSTS",
  "INCLUDE_COORDS",
  "INCLUDE_DEFAULT",
  "INCLUDE_DRAWING_OBJECTS",
  "INCLUDE_MAPPING",
  "INCLUDE_RXN_CODE_ONLY",
  "MOLECULE_DELIMITER",
  "OBJECT_DELIMITER",
  "PRODUCT_IDENTIFIER",
  "RETAIN_REACTANT_AND_PRODUCT_ORDER",
  "decode",
  "encode",
]
`;

exports[`class prototypes > static properties of Reactor 1`] = `[]`;

exports[`class prototypes > static properties of Resources 1`] = `
[
  "register",
  "registerFromNodejs",
  "registerFromUrl",
]
`;

exports[`class prototypes > static properties of SDFileParser 1`] = `[]`;

exports[`class prototypes > static properties of SSSearcher 1`] = `[]`;

exports[`class prototypes > static properties of SSSearcherWithIndex 1`] = `
[
  "bitCount",
  "getHexStringFromIndex",
  "getIndexFromHexString",
  "getKeyIDCode",
  "getSimilarityAngleCosine",
  "getSimilarityTanimoto",
]
`;

exports[`class prototypes > static properties of SmilesParser 1`] = `[]`;

exports[`class prototypes > static properties of ToxicityPredictor 1`] = `
[
  "RISK_HIGH",
  "RISK_LOW",
  "RISK_NAMES",
  "RISK_NO",
  "RISK_UNKNOWN",
  "TYPE_IRRITANT",
  "TYPE_MUTAGENIC",
  "TYPE_REPRODUCTIVE_EFFECTIVE",
  "TYPE_TUMORIGENIC",
]
`;

exports[`class prototypes > static properties of Transformer 1`] = `[]`;

exports[`class prototypes > static properties of Util 1`] = `
[
  "getHoseCodesFromDiastereotopicID",
]
`;

exports[`class prototypes > static properties of registerCustomElement 1`] = `[]`;

exports[`top-level API 1`] = `
[
  "Canonizer",
  "CanonizerUtil",
  "CanvasEditor",
  "ConformerGenerator",
  "DrugScoreCalculator",
  "DruglikenessPredictor",
  "ForceFieldMMFF94",
  "Molecule",
  "MoleculeProperties",
  "Reaction",
  "ReactionEncoder",
  "Reactor",
  "Resources",
  "RingCollection",
  "SDFileParser",
  "SSSearcher",
  "SSSearcherWithIndex",
  "SmilesParser",
  "ToxicityPredictor",
  "Transformer",
  "Util",
  "registerCustomElement",
  "version",
]
`;
